crates/sui/tests/framework_upgrades/mock_sui_systems/deep_upgrade/sources/sui_system_state_inner.move

```diff
@@ -0,0 +1,205 @@
+        storage_fund: Balance<SUI>,
+        protocol_version: u64,
+        epoch_start_timestamp_ms: u64,
+        epoch_duration_ms: u64,
+        ctx: &mut TxContext,
+    ): SuiSystemStateInner {
+        let validators = new_validator_set(validators, ctx);
+        let system_state = SuiSystemStateInner {
+            epoch: 0,
+            protocol_version,
+            system_state_version: genesis_system_state_version(),
+            validators,
+            storage_fund,
+            parameters: SystemParameters {
+                epoch_duration_ms,
+                extra_fields: bag::new(ctx),
+            },
+            reference_gas_price: 1,
+            safe_mode: false,
+            epoch_start_timestamp_ms,
+            extra_fields: bag::new(ctx),
+        };
+        system_state
+    }
+
+    public(friend) fun advance_epoch(
+        self: &mut SuiSystemStateInnerV3,
+        new_epoch: u64,
+        next_protocol_version: u64,
+        storage_reward: Balance<SUI>,
+        computation_reward: Balance<SUI>,
+        storage_rebate_amount: u64,
+        epoch_start_timestamp_ms: u64,
+    ) : Balance<SUI> {
+        touch_dummy_inactive_validator(self);
+
+        self.epoch_start_timestamp_ms = epoch_start_timestamp_ms;
+        self.epoch = self.epoch + 1;
+        assert!(new_epoch == self.epoch, 0);
+        self.safe_mode = false;
+        self.protocol_version = next_protocol_version;
+
+        balance::join(&mut self.storage_fund, computation_reward);
+        balance::join(&mut self.storage_fund, storage_reward);
+        let storage_rebate = balance::split(&mut self.storage_fund, storage_rebate_amount);
+        storage_rebate
+    }
+
+    public(friend) fun advance_epoch_safe_mode(
+        self: &mut SuiSystemStateInnerV3,
+        new_epoch: u64,
+        next_protocol_version: u64,
+        storage_reward: Balance<SUI>,
+        computation_reward: Balance<SUI>,
+        _storage_rebate: u64,
+        _ctx: &mut TxContext,
+    ) {
+        self.epoch = new_epoch;
+        self.protocol_version = next_protocol_version;
+        self.safe_mode = true;
+        balance::join(&mut self.storage_fund, computation_reward);
+        balance::join(&mut self.storage_fund, storage_reward);
+    }
+
+    public(friend) fun protocol_version(self: &SuiSystemStateInnerV3): u64 { self.protocol_version }
+    public(friend) fun system_state_version(self: &SuiSystemStateInnerV3): u64 { self.system_state_version }
+    public(friend) fun genesis_system_state_version(): u64 {
+        SYSTEM_STATE_VERSION_V1
+    }
+
+    fun new_validator_set(init_active_validators: vector<Validator>, ctx: &mut TxContext): ValidatorSet {
+        ValidatorSet {
+            active_validators: init_active_validators,
+            inactive_validators: table::new(ctx),
+            extra_fields: bag::new(ctx),
+        }
+    }
+
+    public(friend) fun v1_to_v3(v2: SuiSystemStateInner): SuiSystemStateInnerV3 {
```

Comment:
The base of this upgrade is v1 so the param name should be v1. 
Also I understand that the shallow upgrade version is v2 so this deep one becomes v3 but this is slightly confusing to have `v1_to_v3`. Maybe we name them `SuiSystemStateInnerV2Shallow` and `SuiSystemStateInnerV2Deep` instead?

Reply:
Renamed